viewing_party/party.py

```diff
@@ -1,23 +1,139 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        return {
+        "title": title,
+        "genre": genre, 
+        "rating": rating
+        }
+    else:
+        return None
+        
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data 
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # watchlist = user_data["watchlist"]
+    # watched = user_data["watched"]
+
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
+
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    list_of_ratings = []
+    for movie in user_data["watched"]:
+        list_of_ratings.append(movie["rating"])

```

Comment:
Consider using list comprehensions syntax (covered as a lightning talk topic), to append a set of values very concisely.

The following would initialize `list_of_ratings` to be a list containing the ratings of each movie from the watched list.
```py
    list_of_ratings = [movie["rating"] for movie in user_data["watched"]]
```

The general pattern lets us take some code like
```py
new_list = []
for item in old_list:
    if should_process(item):
        new_list.append(process(item))
```

and write it like:
```py
new_list = [process(item) for item in old_list if should_process(item)]
```